package/src/skia/Image/useImage.ts

```diff
@@ -1,9 +1,28 @@
-import type { DataSource } from "../Data/Data";
+import { useCallback } from "react";
+
+import type { DataSource, Data } from "../Data/Data";
 import { useRawData } from "../Data/Data";
 import { Skia } from "../Skia";
 
 /**
  * Returns a Skia Image object
  * */
-export const useImage = (source: DataSource) =>
-  useRawData(source, Skia.MakeImageFromEncoded);
+export const useImage = (
+  source: DataSource,
+  onError?: (err: Error) => void
+) => {
+  const factory = useCallback(
+    (data: Data) => {
+      try {
+        return Skia.MakeImageFromEncoded(data);
```

Comment:
Should we use the try catch in useRawData? any `factory` may throw and we need to handle it the same way

Reply:
I was considering this, but decided to include it in the factory since `useRawData` uses the factory in two different places, and it doesn't know about what kind of errors the factory might throw.